File: C++/login/loginmanager.cpp
#include "loginmanager.hpp"
#include "thirdpartystrategy.hpp"
#include "usernamepasswordstrategy.hpp"
#include "emailotpstrategy.hpp"
#include "../common_include_files.hpp"

LoginManager::LoginManager() : strategy(nullptr) {}

bool LoginManager::performLogin() {
    cout << "Attempting " << strategy->getName() << " login..." << endl;
    bool loginSuccess = strategy->login();
    if (loginSuccess) {
        cout << "Login successful!" << endl;
    }
    else {
        cout << "Login failed." << endl;
    }
    return loginSuccess;
}

void LoginManager::SetLoginStrategy(int loginstrategy) {
    if (loginstrategy == USERNAME_PASSWORD) {
        strategy = new UsernamePasswordStrategy();
    }
    else if (loginstrategy == LOGIN_STRATEGY::EMAIL_OTP) {
        strategy = new EmailOtpStrategy();
    }
    else if (loginstrategy == LOGIN_STRATEGY::THIRD_PARTY) {
        strategy = new ThirdPartyStrategy();
    }
}

File: JAVA/Login/LoginManager.java
public class LoginManager {
    private LoginStrategy strategy;

    public LoginManager() {
        this.strategy = null;
    }

    public boolean performLogin() {
        System.out.println("Attempting " + strategy.getName() + " login...");
        boolean loginSuccess = strategy.login();
        if (loginSuccess) {
            System.out.println("Login successful!");
        } else {
            System.out.println("Login failed.");
        }
        return loginSuccess;
    }

    public void setLoginStrategy(int loginStrategy) {
        if (loginStrategy == LoginStrategyEnum.USERNAME_PASSWORD.ordinal()) {
            strategy = new UsernamePasswordStrategy();
        } else if (loginStrategy == LoginStrategyEnum.EMAIL_OTP.ordinal()) {
            strategy = new EmailOtpStrategy();
        } else if (loginStrategy == LoginStrategyEnum.THIRD_PARTY.ordinal()) {
            strategy = new ThirdPartyStrategy();
        }
    }

    public void displayLoginMethods() {
        // Implement the method to display login methods as needed
    }
}
